packages/gatsby-source-filesystem/src/create-remote-file-node.js

```diff
@@ -199,7 +200,7 @@ async function processRemoteNode({
   // See if there's response headers for this url
   // from a previous request.
   const cachedHeaders = await cache.get(cacheId(url))
-  const headers = {}
+  headers = { headers: { ...headers } }
```

Comment:
can we maybe rename top level - because this line is really confusing when I first read it:
```suggestion
  const requestArgs = { headers: { ...headers } }
```
or something like that? (and of course adjust when needed afterwards)

this actually also made me question if our current code for `If-None-Match` header do anything it all given that it's passed in top level for `got` request (and not in `headers`) - but this is definitely not in scope of this PR

Reply:
also - do we need to create extra object copy of headers object (`{...headers}`)? 

wouldn't:
```
const requestArgs = { headers }
```
work?

Reply:
I think we can do  it. I will modify my changes.

Reply:
yes, current code for If-none-match not caching header correctly.

Reply:
I modify the code as you suggested. can you review this now

Reply:
@pieh But with just  { headers } isn’t there danger that the code will later start mutating the headers causing unexpected side-effects and it will be up to caller to ensure that it only gives copy of its configured headers when calling.